doc/tutorials/agl-vms.rst

```diff
@@ -302,139 +259,139 @@ Setting up AGLs
+      offline_path="/sys/class/acrn/acrn_hsm"
+      fi
+
+      function launch_clear()
       {
-        vm_name=vm$1
-
-        #check if the vm is running or not
-        vm_ps=$(pgrep -a -f acrn-dm)
-        result=$(echo $vm_ps | grep "${vm_name}")
-        if [[ "$result" != "" ]]; then
-          echo "$vm_name is running, can't create twice!"
-          exit
-        fi
-
-        # create a unique tap device for each VM
-        tap=tap2
-        tap_exist=$(ip a | grep "$tap" | awk '{print $1}')
-        if [ "$tap_exist"x != "x" ]; then
-          echo "tap device existed, reuse $tap"
-        else
-          ip tuntap add dev $tap mode tap
-        fi
-
-        # if acrn-br0 exists, add VM's unique tap device under it
-        br_exist=$(ip a | grep acrn-br0 | awk '{print $1}')
-        if [ "$br_exist"x != "x" -a "$tap_exist"x = "x" ]; then
-          echo "acrn-br0 bridge aleady exists, adding new tap device to it..."
-          ip link set "$tap" master acrn-br0
-          ip link set dev "$tap" down
-          ip link set dev "$tap" up
-        fi
-
-        #for memsize setting
-        mem_size=2048M
-
-        acrn-dm -A -m $mem_size -c $2 -s 0:0,hostbridge -s 1:0,lpc -l com1,stdio \
-          -s 2,pci-gvt -G "$3" \
-          -s 5,virtio-console,@pty:pty_port \
-          -s 6,virtio-hyper_dmabuf \
-          -s 3,virtio-blk,/root/agl_ivi.wic \
-          -s 4,virtio-net,$tap \
-          -s 7,xhci,1-4 \
-          -k /root/bzImage-4.19.0-uos \
-          -B "root=/dev/vda2 rw rootwait maxcpus=$2 nohpet console=tty0 console=hvc0 \
-          console=ttyS0 no_timer_check ignore_loglevel log_buf_len=16M \
-          consoleblank=0 tsc=reliable i915.avail_planes_per_pipe=$4 \
-          i915.enable_hangcheck=0 i915.nuclear_pageflip=1 i915.enable_guc_loading=0 \
-          i915.enable_guc_submission=0 i915.enable_guc=0" $vm_name
-      }
+      mac=$(cat /sys/class/net/e*/address)
+      vm_name=vm$1
+      mac_seed=${mac:9:8}-${vm_name}
 
+      #check if the vm is running or not
+      vm_ps=$(pgrep -a -f acrn-dm)
+      result=$(echo $vm_ps | grep -w "${vm_name}")
+      if [[ "$result" != "" ]]; then
+        echo "$vm_name is running, can't create twice!"
+        exit
+      fi
 
-      # offline SOS CPUs except BSP before launch UOS
-      for i in `ls -d /sys/devices/system/cpu/cpu[2-99]`; do
-        online=`cat $i/online`
-        idx=`echo $i | tr -cd "[2-99]"`
-        echo cpu$idx online=$online
-        if [ "$online" = "1" ]; then
-          echo 0 > $i/online
-          echo $idx > /sys/class/vhm/acrn_vhm/offline_cpu
-        fi
-      done
+      #logger_setting, format: logger_name,level; like following
+      logger_setting="--logger_setting console,level=4;kmsg,level=3"
 
-      launch_agl 1 1 "64 448 8" 0x000F00 agl
+      #for memsize setting
+      mem_size=2048M
 
-#. Create the ``launch_rse.sh`` script for the AGL RSE VM with this
-   content::
+      acrn-dm -A -m $mem_size -c $2 -s 0:0,hostbridge \
```

Comment:
Just double-checking: do we still have the `-c` option in that version of `acrn-dm`?

Reply:
yes, it's still have in that version